### PR TITLE
feat: Make PostgreSQL destination use streaming batch writer (writes per table)

### PR DESCRIPTION
**Pull request by candiduslynx**
~Depends on https://github.com/cloudquery/plugin-sdk/pull/1131~

### CODE REVIEW

File: plugins/destination/postgresql/client/client.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (c *Client) Write(ctx context.Context, res <-chan message.WriteMessage) err
 	return c.writer.Write(ctx, res)
 }
 
-func (c *Client) Close(ctx context.Context) error {
+func (c *Client) Close(context.Context) error {
 	var err error
```
Comment (disq) [inline plugins/destination/postgresql/client/client.go:110]: should do `c.writer.Close(ctx)` here

File: plugins/destination/postgresql/client/insert.go [hunk 2]
```diff
@@ -13,61 +13,36 @@ import (
...
+				sql = c.upsert(table)
+			} else {
+				sql = c.insert(table)
+			}
+		}
+
+		for _, row := range transformValues(msg.Record) {
+			batch.Queue(sql, row...)
 		}
 	}
```
Comment (disq) [inline plugins/destination/postgresql/client/insert.go:47]: batch should be sent on channel close

Comment (candiduslynx) [reply plugins/destination/postgresql/client/insert.go:47]: well, batch.Queue only adds the rows, the sending is done after iterating over mesages

File: plugins/destination/postgresql/client/list_tables.go [hunk 3]
```diff
@@ -53,9 +53,8 @@ ORDER BY
 	table_name ASC, ordinal_position ASC;
 `
 
-func (c *Client) listTables(ctx context.Context, include, exclude []string) (schema.Tables, error) {
-	var tables schema.Tables
-	whereClause := c.whereClause(include, exclude)
+func (c *Client) getDBTable(ctx context.Context, name string) (*schema.Table, error) {
+	whereClause := "AND pg_class.relname = '" + inClause(name) + "'"
```
Comment (disq) [inline plugins/destination/postgresql/client/list_tables.go]:
```suggestion
	whereClause := "AND pg_class.relname IN " + inClause(name)
```
I don't think this complex code is necessary as the migrate table is invoked separately for each table

File: plugins/destination/postgresql/pgarrow/to_arrow.go [hunk 4]
```diff
@@ -45,6 +45,10 @@ func Pg10ToArrow(t string) arrow.DataType {
 		return arrow.PrimitiveTypes.Int32
 	case "bigint", "int8":
 		return arrow.PrimitiveTypes.Int64
+	case "numeric(20,0)":
```
Comment (disq) [inline plugins/destination/postgresql/pgarrow/to_arrow.go:48]: Why not check for `strings.HasPrefix(t, "numeric(") && strings.HasSuffix(t, ",0)")`? (numerics with float components should still remain strings)

File: plugins/destination/postgresql/client/client.go [hunk 5]
```diff
@@ -107,15 +107,14 @@ func (c *Client) Write(ctx context.Context, res <-chan message.WriteMessage) err
 }
 
 func (c *Client) Close(ctx context.Context) error {
-	var err error
 	if c.conn == nil {
 		return fmt.Errorf("client already closed or not initialized")
 	}
 	if c.conn != nil {
```
Comment (disq) [inline plugins/destination/postgresql/client/client.go:113]:
```suggestion
	err := c.writer.Close(ctx)
	if c.conn != nil {
```

**Review by disq — APPROVED** [review]

**Pull request closed without merging**